### PR TITLE
Update composite documentation to be more explicit

### PR DESCRIPTION
**Pull request by NBParis**
### What does this PR do?
Gives a fresh update to the composite documentation that had too many examples and those were not very clear.

### Motivation
Composite behaviour is hard to understand for our users as a lot of important information are lost in the middle of the page. The goal was therefore to better highlight the key behaviour with simplified examples.

### Preview

https://docs-staging.datadoghq.com/nils/composite-monitor-update/monitors/monitor_types/composite/#overview

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/monitors/monitor_types/composite.md [hunk 1]
```diff
@@ -93,84 +95,80 @@ This section uses examples to show how trigger conditions are computed, and the
...
-
-If `A` is no alert-worthy, `!A` is `Alert`
-
+* If `A && B` is alert-worthy, the result is the **least** severe status between A and B.
+* If `A || B` is alert-worthy, the result is the **most** severe status between A and B.
+* If `A` is `No Data`, `!A` is `No Data`
+* If `A` is alert-worthy, `!A` is `OK`
+* If `A` is no alert-worthy, `!A` is `Alert`
 
 Consider a composite monitor that uses three individual monitors (`A`, `B`, `C`) and a trigger condition `A && B && C`. The following table shows the resulting status of the composite monitor given different statuses for its individual monitors (alert-worthiness is indicated with T or F):
```
Comment (dussault-antoine) [inline content/en/monitors/monitor_types/composite.md:115]:
```suggestion
Consider a composite monitor that uses two individual monitors `A` and `B`. The following table shows the resulting status of the composite monitor given the trigger condition (`&&` or `||`), and the different statuses for its individual monitors (alert-worthiness is indicated with T or F):
```

**Review by dussault-antoine — CHANGES_REQUESTED** [review]
Table below suggestion is going through examples of a composite monitor relying on 2 monitors, not 3 monitors

File: content/en/monitors/monitor_types/composite.md [hunk 2]
```diff
@@ -75,11 +76,12 @@ a || b && !c
 
 For detailed instructions on the **Say what's happening** and **Notify your team** sections, see the [Notifications][2] page.
 
-## API
+### API
 
-When using the API, a non-composite monitor's query can encapsulate a metric, tags, aggregation function like `avg`, group-by clause, etc. A composite monitor's query is defined in terms of its constituent monitors using monitor IDs.
+When using the API, a composite monitor's query is defined in terms of its constituent monitors using monitor IDs.
```
Comment (apigirl) [inline content/en/monitors/monitor_types/composite.md:81]:
```suggestion
When using the API, a composite monitor's query is defined in terms of its constituent monitors using monitor IDs.
```
I am not sure what you mean here by constituent monitors? Dependent monitors? Related monitors?

File: content/en/monitors/monitor_types/composite.md [hunk 3]
```diff
@@ -93,84 +95,80 @@ This section uses examples to show how trigger conditions are computed, and the
 
 ### Computing trigger conditions
 
-Datadog computes `A && B && C` in the way you would expect, but which monitor statuses are considered alert-worthy? There are six different statuses a composite monitor considers:
+There are 4 different statuses a composite monitor considers and 3 of them are considered as alert-worthy:
```
Comment (apigirl) [inline content/en/monitors/monitor_types/composite.md:98]:
```suggestion
There are 4 different statuses a composite monitor and 3 of them are considered as alert-worthy:
```

**Review by apigirl — COMMENTED** [review]
just some initial feedback

File: content/en/monitors/monitor_types/composite.md (hunk 2 above)
Comment (NBParis) [reply content/en/monitors/monitor_types/composite.md:81]: Used the same word as before but we definitely intent to say "Child monitor" or sub-monitor here.

Comment (NBParis) [inline content/en/monitors/monitor_types/composite.md:81]:
```suggestion
When using the API, a composite monitor's query is defined in terms of its sub-monitors using monitor IDs.
```

File: content/en/monitors/monitor_types/composite.md [hunk 4]
```diff
@@ -93,84 +95,80 @@ This section uses examples to show how trigger conditions are computed, and the
 
 ### Computing trigger conditions
 
-Datadog computes `A && B && C` in the way you would expect, but which monitor statuses are considered alert-worthy? There are six different statuses a composite monitor considers:
+There are 4 different statuses a composite monitor and 3 of them are considered as alert-worthy:
```
Comment (apigirl) [inline content/en/monitors/monitor_types/composite.md:98]:
```suggestion
There are 4 different statuses a composite monitor and 3 of them are considered alert-worthy:
```

File: content/en/monitors/monitor_types/composite.md [hunk 5]
```diff
@@ -93,84 +95,80 @@ This section uses examples to show how trigger conditions are computed, and the
...
-| Warn (T)    | Ok (F)      | Warn (T)                                       | {{< X >}}        |
-| OK (F)      | No Data (T) | No Data (T) _(if Notify No Data is enabled)_   | {{< X >}}        |
-| OK (F)      | No Data (F) | OK (F)      _(if Notify No Data is disabled)_  |                  |
+* If `A && B` is alert-worthy, the result is the **least** severe status between A and B.
+* If `A || B` is alert-worthy, the result is the **most** severe status between A and B.
+* If `A` is `No Data`, `!A` is `No Data`
+* If `A` is alert-worthy, `!A` is `OK`
+* If `A` is no alert-worthy, `!A` is `Alert`
+
+Consider a composite monitor that uses two individual monitors `A` and `B`. The following table shows the resulting status of the composite monitor given the trigger condition (`&&` or `||`), and the different statuses for its individual monitors (alert-worthiness is indicated with T or F):
```
Comment (apigirl) [inline content/en/monitors/monitor_types/composite.md:115]:
```suggestion
Consider a composite monitor that uses two individual monitors: `A` and `B`. The following table shows the resulting status of the composite monitor given the trigger condition (`&&` or `||`), and the different statuses for its individual monitors (alert-worthiness is indicated with T or F):
```

File: content/en/monitors/monitor_types/composite.md [hunk 6]
```diff
@@ -93,84 +95,80 @@ This section uses examples to show how trigger conditions are computed, and the
...
+| No Data (T) | Warn (T)    | `A && B`    | False            | Last known       |                  |
+| No Data (T) | Warn (T)    | `A || B`    | False            | Warn (T)         | {{< X >}}        |
+| No Data (T) | OK (F)      | `A && B`    | False            | OK (F)           |                  |
+| No Data (T) | OK (F)      | `A || B`    | False            | Last known       |                  |
+| No Data (T) | OK (F)      | `A && B`    | True             | OK (F)           |                  |
+| No Data (T) | OK (F)      | `A || B`    | True             | No Data (T)      | {{< X >}}        |
+| No Data (T) | No Data (T) | `A && B`    | True             | No Data (T)      | {{< X >}}        |
+| No Data (T) | No Data (T) | `A || B`    | True             | No Data (T)      | {{< X >}}        |
+
+**Note**: When the composite has `notify_no_data` to false and the result of the evaluation of the sub-monitors should end up on a `No Data` status for the composite, the composite uses the last known state instead.
```
Comment (apigirl) [inline content/en/monitors/monitor_types/composite.md:134]:
```suggestion
**Note**: When the composite has `notify_no_data` to false, and the result of the evaluation of the sub-monitors should end up on a `No Data` status for the composite, the composite uses the last known state instead. 
```

**Review by apigirl — APPROVED** [review]

**Pull request merged**